k8s.gcr.io/lib.sh

```diff
@@ -0,0 +1,279 @@
+        echo "gcs_bucket_for(repo, [region]) requires 1 or 2 arguments" > /dev/stderr
+        return 1
+    fi
+    repo="$1"
+    region="${2:-}"
+
+    if [ -z "${region}" ]; then
+        echo "gs://artifacts.${repo}.appspot.com"
+    else
+        echo "gs://${region}.artifacts.${repo}.appspot.com"
+    fi
+}
+
+# Get the GCR host name for a region
+# $1: The GCR region
+function gcr_host_for() {
+    if [ $# != 1 ]; then
+        echo "gcr_host_for(region) requires 1 argument" > /dev/stderr
+        return 1
+    fi
+    region="$1"
+
+    if [ -z "${region}" ]; then
+        echo "gcr.io"
+    else
+        echo "${region}.gcr.io"
+    fi
+}
+
+# Get the service account email for a given short name
+# $1: The GCP project
+# $2: The name
+function svc_acct_for() {
+    if [ $# != 2 -o -z "$1" -o -z "$2" ]; then
+        echo "svc_acct_for(project, name) requires 2 arguments" > /dev/stderr
+        return 1
+    fi
+    project="$1"
+    name="$2"
+
+    echo "${name}@${project}.iam.gserviceaccount.com"
+}
+
+# Ensure that a project exists in our org.
+# $1: The GCP project
+function ensure_project() {
+    if [ $# != 1 -o -z "$1" ]; then
+        echo "ensure_project(project) requires 1 argument" > /dev/stderr
+        return 1
+    fi
+    project="$1"
+
+    if ! gcloud projects describe "${project}" >/dev/null 2>&1; then
+        gcloud projects create "${project}" \
+            --organization "${GCP_ORG}"
+    else
+        o=$(gcloud projects \
+                describe "${project}" \
+                --flatten='parent[]' \
+                --format='csv[no-heading](type, id)' \
+                | grep ^organization \
+                | cut -f2 -d,)
+        if [ "$o" != "${GCP_ORG}" ]; then
+            echo "project ${project} exists, but not in our org: got ${o}" > /dev/stderr
+            return 2
+        fi
+    fi
+}
+
+# Link a project to our billing account
+# $1: The GCP project
+function ensure_billing() {
+    if [ $# != 1 -o -z "$1" ]; then
+        echo "ensure_billing(project) requires 1 argument" > /dev/stderr
+        return 1
+    fi
+    project="$1"
+
+    gcloud beta billing projects link "${project}" \
+        --billing-account "${GCP_BILLING}"
+}
+
+# Enable an API
+# $1: The GCP project
+# $2: The API (e.g. containerregistry.googleapis.com)
+function enable_api() {
+    if [ $# != 2 -o -z "$1" -o -z "$2" ]; then
+        echo "enable_api(project, api) requires 2 arguments" > /dev/stderr
+        return 1
+    fi
+    project="$1"
+    api="$2"
+
+    gcloud --project "${project}" services enable "${api}"
+}
+
+# Ensure the bucket backing the repo exists and is world-readable
+# $1: The GCP project
+function ensure_repo() {
+    if [ $# -lt 1 -o $# -gt 2 -o -z "$1" ]; then
+        echo "ensure_repo(project, [region]) requires 1 or 2 arguments" > /dev/stderr
+        return 1
+    fi
+    project="$1"
+    region="${2:-}"
+
+    if ! gsutil ls $(gcs_bucket_for "${project}" "${region}") >/dev/null 2>&1; then
```

Comment:
Please separate the call to `gcs_bucket_for` to its own line.

Reply:
ACK